modules/base/rendering/renderableswitch.cpp

```diff
@@ -0,0 +1,141 @@
+ * Copyright (c) 2014-2025                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rendering/renderableswitch.h>
+
+#include <modules/base/basemodule.h>
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <ghoul/filesystem/file.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/io/texture/texturereader.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/misc/crc32.h>
+#include <ghoul/misc/profiling.h>
+#include <ghoul/opengl/texture.h>
+#include <fstream>
+#include <optional>
+
+ //renderableSelectRenderable
+ //add by distance to the end
+ //renderableSelectorByDistance
+ //RenderableSwitcherByDistance
+namespace {
+    constexpr openspace::properties::Property::PropertyInfo AutoScaleInfo = {
+        "AutoScale",
+        "Auto Scale",
+        "When true, the plane will automatically adjust in size to match the aspect "
+        "ratio of the content. Otherwise it will remain in the given size."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo DistanceThresholdInfo = {
+        "DistanceThreshold",
+        "Distance Threshold",
+        "Threshold for when the switch happens between the two renderables. "
+    };
+
+    struct [[codegen::Dictionary(RenderableSwitch)]] Parameters {
```

Comment:
If you add a documentation above this struct, it will show up as a general explanation what this Renderable does to the user. See https://github.com/OpenSpace/OpenSpace/blob/master/modules/base/rendering/renderablenodearrow.cpp#L208-L215 for example